schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,318 @@
+  for the schema group, the format MUST be specified at the schema-level. This
+  specification does not mandate support for particular formats.
+- Constraints:
+  - REQUIRED if not set at the schema group level.
+  - MUST be null (not present) if set at the schema group level.
+  - MUST be a non-empty string, if present.
+- Examples:
+  - "avro"
+  - "json-schema"
+  - "xsd11"
+
+#### description
+
+- Type: `String`
+- Description: Explains the purpose of the schema.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "This schema describes the toppings for a Pizza."
+
+  #### createdtimeutc
+
+- Type: `Timestamp`
+- Description: Instant when the schema was added to the registry.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+#### updatedtimeutc
+
+- Type: `Timestamp`
+- Description: Instant when the schema was last updated
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+### 2.2.3 Schema version attributes
+
+A schema version is a document. The "body" of a schema version MAY be a text
+document or binary stream. An implementation SHOULD validate whether a
+schema version is valid according to the rules of its format, for instance
+whether it is a valid Avro schema document when the format is Apache Avro.
+
+Within the scope of the schema set, the version is identified by the combination
+of a version number and an optional format identifier. The schema version MAY
+also have an additional, optional unique identifier within the scope of the
+registry.
+
+#### version
+
+- Type: `Integer`
+- Description: The version of the schema. This is a simple counter and tracks
+  the version in the scope of this schema within the schema group. The schema
+  document content MAY indicate a different versioning scheme.
+- Constraints:
+  - REQUIRED
+  - Assigned by server.
+- Examples:
+  - 1
+  - 2
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema document uniquely without requiring
+  other qualifiers.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+- Examples:
+  - { ... guid ... }
+
+#### description
+
+- Type: `String`
+- Description: Explains details of the schema version.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "This version adds support for different types of Pizza crust."  
+
+#### createdtimeutc
+
+- Type: `Timestamp`
+- Description: Instant when the schema version was added to the registry.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+## 3. HTTP ("REST") API 
+
+This section informally describes the HTTP API binding of this schema registry.
+The formal definition is the [OpenAPI document](./schemaregistry.yaml) that is
+part of this specification.
+
+### 3.1. Path hierarchy
+
+Schema groups contain schemas and those contain schema versions, which are the
+documents required for serialization or validation.
+
+These dependencies are reflected in the path structure:
+
+`[/schemagroups]/{group-id}/schemas/{schema-id}/versions/{version}`
```

Comment:
Not sure if it matters, but the use of the word `id` got me wondering if it should be `name` instead. While the semantics would be the same either way, all examples of these IDs appear to be more like human friendly names rather than IDs (e.g GUIDs). So while it's not totally human friendly (eg no spaces, etc...), if we expect people to use meaningful "words" and not "random chars", then perhaps `name` would help guide them in that direction.

Just a thought

Reply:
Do we need to add something here to indicate that the `/versions/{version}` part is optional?
`[/schemagroups]/{group-id}/schemas/{schema-id}[/versions/{version}]`
or text?